src/main/app/controller/lists/ContactTypesController.ts

```diff
@@ -0,0 +1,163 @@
+        ? (ct: ContactType): Promise<ContactType> => req.scope.cradle.api.updateContactType(ct)
+        : (ct: ContactType): Promise<ContactType> => req.scope.cradle.api.createContactType(ct);
+
+      await api(contactType)
+        .then(async () => await this.renderAll(req, res, true))
+        .catch((reason: AxiosError) => {
+          const error = reason.response?.status === 409
+            ? this.contactTypeAlreadyExistsError
+            : this.updateContactTypeError;
+          this.renderContactType(res, contactType, false, [{text: error }]);
+        });
+
+    } else {
+      this.renderContactType(res, contactType, false, validationResult.errors, false, validationResult.nameValid);
+    }
+  }
+
+  public async getDeleteConfirmation(req: AuthedRequest, res: Response): Promise<void> {
+    const idToDelete = req.params.id;
+    const name = req.query?.type?.toString();
+    this.renderDeleteConfirmation(res, name, idToDelete);
+  }
+
+  public async delete(req: AuthedRequest, res: Response): Promise<void> {
+    const idToDelete = req.params.id;
+
+    await req.scope.cradle.api.deleteContactType(idToDelete)
+      .then(() => this.renderAll(req, res, true))
+      .catch((reason: AxiosError) => {
+        const error = reason.response?.status === 409
+          ? this.contactTypeInUseError
+          : this.deleteError;
+        this.renderAll(req, res, false, [{ text: error }]);
+      });
+  }
+
+  private async renderAll(req: AuthedRequest, res: Response, updated = false, errors: { text: string }[] = []): Promise<void> {
+    let contactTypes: ContactType[] = [];
+    await req.scope.cradle.api.getContactTypes()
+      .then((value: ContactType[]) => contactTypes = value)
+      .catch(() => errors.push({ text: this.getContactTypesError }));
+
+    const pageData = {
+      errors: errors,
+      updated: updated,
+      contactTypes: contactTypes?.sort((a, b) => a.type.localeCompare(b.type)),
+      editUrl: this.editContactTypeUrl,
+      deleteUrl: this.deleteConfirmUrl
+    };
+    res.render('lists/tabs/contactTypes/contactTypeContent', pageData);
+  }
+
+  private renderContactType(
+    res: Response,
+    contactType: ContactType,
+    updated: boolean,
+    errorMsgs: { text: string }[] = [],
+    fatalError = false,
+    nameValid = true): void {
+
+    const pageData = {
+      contactType: contactType,
+      updated: updated,
+      errors: errorMsgs,
+      nameValid: nameValid,
+      fatalError: fatalError
+    };
+    res.render('lists/tabs/contactTypes/editContactType', pageData);
+  }
+
+  private validateContactType(contactType: ContactType):
+    { nameValid: boolean; errors: { text: string }[]} {
+    let nameValid = true;
+    const validationErrors: { text: string }[] = [];
+
+    if (!contactType.type?.trim()) {
+      nameValid = false;
+      validationErrors.push({ text: this.nameRequiredError });
+    }
+
+    return { nameValid: nameValid, errors: validationErrors};
```

Comment:
Could this be shortened to: 

  private validateContactType(contactType: ContactType):
    { nameValid: boolean; errors: { text: string }[]} {

    return !contactType.type?.trim() ? { nameValid: false, errors: [{ text: this.nameRequiredError }]
        : { nameValid: true, errors: [] };
  }

Reply:
leaving it for now easier to add more cases